opentelemetry-sdk/benches/metrics_counter.rs

```diff
@@ -127,6 +144,54 @@ fn counter_add_unsorted(c: &mut Criterion) {
     });
 }
 
+fn counter_add_sorted_with_non_static_values(c: &mut Criterion, attribute_values: [String; 10]) {
+    let counter = create_counter("Counter_Add_Sorted_With_Non_Static_Values");
+    c.bench_function("Counter_Add_Sorted_With_Non_Static_Values", |b| {
+        b.iter_batched(
+            || {
+                // 4*4*10*10 = 1600 time series.
+                CURRENT_RNG.with(|rng| {
+                    let mut rng = rng.borrow_mut();
+                    [
+                        rng.gen_range(0..4),
+                        rng.gen_range(0..4),
+                        rng.gen_range(0..10),
+                        rng.gen_range(0..10),
+                    ]
+                })
+            },
+            |rands| {
+                let index_first_attribute = rands[0];
+                let index_second_attribute = rands[1];
+                let index_third_attribute = rands[2];
+                let index_fourth_attribute = rands[3];
+                counter.add(
+                    1,
+                    &[
+                        KeyValue::new(
+                            "attribute1",
+                            attribute_values[index_first_attribute].as_str().to_owned(),
```

Comment:
With #2089, we would be able to avoid the string allocation
```suggestion
                            attribute_values[index_first_attribute].as_str(),
```